packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -20,44 +20,121 @@ trait Pool<B> {
     fn get_batches_mut(&mut self) -> &mut Vec<Arc<RwLock<B>>>;
 }
 
-/// A pool is an ordered collection of batches, where each group of agents within a simulation run
-/// is associated to a batch in the pool.
+/// A pool is an ordered collection of batches.
+///
+/// Each group of agents within a simulation run is associated to a [`Batch`] in the pool. Each
+/// [`Batch`] in a pool can either be borrowed as shared ([`read_proxies()`] and
+/// [`partial_read_proxies()`] returning [`PoolReadProxy`]) or mutable reference
+/// ([`write_proxies()`] and [`partial_write_proxies()`] returning [`PoolReadProxy`]).
+///
+/// [`read_proxies()`]: Self::read_proxies
+/// [`partial_read_proxies()`]: Self::partial_read_proxies
+/// [`write_proxies()`]: Self::write_proxies
+/// [`partial_write_proxies()`]: Self::partial_write_proxies
 pub trait BatchPool<B: Batch>: Send + Sync {
+    /// Creates a new pool from [`Batches`].
+    ///
+    /// Because of the way, `BatchPools` are organized, it's required, that the [`Batch`]es are
+    /// stored inside an [`RwLock`] behind an [`Arc`]. This is subject to change.
     fn new(batches: Vec<Arc<RwLock<B>>>) -> Self;
 
+    /// Creates a new empty pool.
     fn empty() -> Self
     where
         Self: Sized,
     {
         Self::new(Vec::new())
     }
 
+    /// Returns the number of [`Batch`]es inside this pool.
     fn len(&self) -> usize;
 
+    /// Returns if there are no [`Batch`]es in this pool.
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
 
+    /// Adds a [`Batch`] at the end of this pool.
+    ///
+    /// Note, that unless in [`new()`](Self::new) this does not require the [`Batch`] to be wrapped
+    /// inside of [`Arc`]`<RwLock<B>>`.
     fn push(&mut self, batch: B);
 
+    /// Removes and returns the `element` at position `index` within the pool, shifting all
+    /// elements after it to the left and returns it as [`BatchReadProxy`].
```

Comment:
So writing these docs have made me question this signature and interface in general.

As you've suggested with the rename to `View` proxies are supposed to kind of be references to the element. And I don't think it makes sense to return a 'reference' with a `remove` method. I think we should be returning the arc, or something else because we're conceptually giving ownership of the batch.

If these are only used to get the id's of deleted batches, e.g. for some `remove_ids` field in `update_agent_snapshot` , then why don't we just return the id of the batch we deleted. (And in fact, what would be better, is if this method verifies we have exclusive access to the arc of the batch when we drop it, to ensure the batch is actually dropped)

Reply:
Will be done in a [follow up task](https://app.asana.com/0/1201793759102797/1201819939567711/f) (_internal_).

Reply:
We've made a PR to address this #315